src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,195 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private T item;
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> prev;
+        private Node<T> next;
+
+        Node(Node<T> prev, T element, Node<T> next) {
+            this.item = element;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
+    {
+        Node<T> head = null;
+        Node<T> tail = null;
+    }
+
+    public MyLinkedList(T item) {
+        this.item = item;
+    }
+
+    public MyLinkedList(){
+    }
+
+    public void checkIndex(int index) {
+        if (index >= size || index < 0) {
+            throw new IndexOutOfBoundsException("exception");
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> node = new Node<>(null, value, null);
+        if (head == null) {
+            head = tail = node;
+            head.prev = null;
+            tail.next = null;
```

Comment:
if your Node have such structure new Node<>(null, value, null) so do we need to make links null?
And why do we use tail here?